benchmark-service/benchmark/benchmark.py

```diff
@@ -215,27 +217,59 @@ async def compare(request, userdata):  # pylint: disable=unused-argument
     return await render_template('benchmark', request, userdata, 'compare.html', context)
 
 
-async def query_github(app):
-    global START_POINT
+async def update_commits(app):
+    global benchmark_data
     github_client = app['github_client']
-    request_string = f'/repos/hail-is/hail/commits?since={START_POINT}'
+    batch_client = app['batch_client']
+    gs_reader = app['gs_reader']
 
-    data = await github_client.getitem(request_string)
+    request_string = f'/repos/hail-is/hail/commits?since={START_POINT}'
+    gh_data = await github_client.getitem(request_string)
     new_commits = []
-    for commit in data:
-        sha = commit.get('sha')
-        new_commits.append(commit)
-        log.info(f'commit {sha}')
-        START_POINT = commit['commit']['author'].get('date')
-        log.info(f'start point is now {START_POINT}')
+    formatted_new_commits = []
+    for gh_commit in gh_data:
+
+        sha = gh_commit.get('sha')
+
+        batches = [b async for b in batch_client.list_batches(q=f'sha={sha} running')]
+        try:
```

Comment:
This is a better way to write this:

```
batches = [b async for b in batch_client.list_batches(q=f'sha={sha} running')]
if len(batches) == 0:
    batch_status = None
else:
    batch = batches[-1]
    batch_status = await batch.status()
```
```

Reply:
I forgot why I was trying to get the status of the last batch